.github/workflows/hls.yml

```
@@ -0,0 +1,47 @@
+name: "Haskell Language Server works"
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+ 
+permissions:
+  contents: read
+  
+jobs:
+  test-hls-works:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+      - name: Install Nix with good defaults
+        uses: input-output-hk/install-nix-action@v20
+        with:
+          extra_nix_config: |
+            trusted-public-keys = cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY= hydra.iohk.io:f/Ea+s+dFdN+3Y/G+FDgSq+a5NEWhJGzdjvKNGv0/EQ=
+            substituters = https://cache.iog.io/ https://cache.nixos.org/
+          nix_path: nixpkgs=channel:nixos-unstable 
+      - name: Cachix install Nix
+        uses: cachix/install-nix-action@v18
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - name: Open nix environment
+        uses: rrbutani/use-nix-shell-action@v1
+      - name: Update dependencies
+        run: cabal update; cabal freeze
+      - name: Obtain GHC version
+        run: |
+          echo "VERSION=$(ghc --version | grep -oE '[^ ]+$')" >> $GITHUB_OUTPUT
+        id: ghc
+      - name: Cache hie-bios and ghcide 
+        uses: actions/cache@v3
```

```suggestion
        uses: actions/cache@v4
```

Why not use the latest version available?